src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig

```diff
@@ -236,6 +236,31 @@
 
         {% if excluded_ajax_paths is defined %}
             if (window.XMLHttpRequest && XMLHttpRequest.prototype.addEventListener) {
+                if (window.fetch) {
+                    var oldFetch = window.fetch;
+                    window.fetch = function () {
+                        var promise = oldFetch.apply(null, arguments);
+                        var stackElement = {
+                            loading: true,
+                            error: false,
+                            url: arguments[0],
+                            method: arguments[1].method,
+                            start: new Date()
+                        };
+                        requestStack.push(stackElement);
+                        promise.then(function (r) {
```

Comment:
what if the promise gets rejected ? The stack element should not stay in loading state in this case